### ISSUE TITLE
Donation Receipts and BCC Issue

### ISSUE THREAD

**Issue by stephaniemcv** [mozilla/donate-wagtail#1565]
**Describe the bug**
Receipts generated by Acoustic Transact are no longer sending emails to receipt-bcc@mozillafoundation.org in addition to the donor's email address. 

On 9/1/21 the receipt-bcc@mozillafoundation.org google group was impacted as part of a Mozilla wide incident and stopped receiving messages of receipts sent from Acoustic Transact. 

Per Jlin on #servicedesk - "we are currently experiencing an issue with sending emails to google groups, with the email bouncing back with an error message. We are investigating it with google support, however, if you can try to resend the email but with only the google group in the send-to (no cc’s of other people or original sender if they are also part of the group) it might go through successfully."

on 9/2/21 Jlin sent a service restoration email that functionality had been restored. He confirmed via slack that this included emails that have a groups email address in the cc or bcc field. Receipts continued to generate in Transact and send to the donor but not to the bcc address.

@stephanielowry and I successfully tested via direct sends from our mozilla inboxes that emails that included the receipt-bcc@mozillafoundation.org in the bcc field are sent to the group. The configuration of the group is still set to "Anyone on the Web Can Post".

Since the XML info to Transact contains the bbc email address, we're wondering if something there was impacted by the 9/1 incident.

**To Reproduce**
1. Give a donation.
2. Go to groups.google.com/
3. Scroll down to and select Receipt BCC - receipt-bcc@mozillafoundation.org (Copies of donation receipts for fraud and donor care support)
4. See if your donation receipt shows up.

**Expected behavior**
A copy of the donation receipt should be findable in the group.

**Screenshots**

![image](https://user-images.githubusercontent.com/11812272/137554061-0b03142f-6045-4268-aa48-d0807f9470a7.png)

**Comment by jzinner** [comment]
Contact #servicedesk to get added to the receipt-bcc@mozillafoundation.org google group

This may work, try adding the email to line 110 tasks.py for bcc?
https://github.com/mozilla/donate-wagtail/blob/e46ef80ec4e87513617539001aec080a95d5d5db/donate/payments/tasks.py#L110

**Comment by jzinner** [comment]
This is a priority for 2021 EOY operations to support the donation campaigns.

**Comment by danielfmiranda** [comment]
<img width="1363" alt="image" src="https://user-images.githubusercontent.com/18314510/142487050-761ec48f-f3b8-45ea-87ca-9eff263f367a.png">

Hey everyone, we have tried updating the BCC field in the donate-stack code like suggested above to send to the mozilla-receipt email adress. However we experienced the error above, and it appears that BCC is not enabled for this campaign. 

Following this, we took a look at the acoustic documentation: https://help.goacoustic.com/hc/en-us/articles/360043609093-Email-administration

And it appears that the acoustic configuration needs to be changed in order for us to add a BCC, or, can we add a BCC on the acoustic side?

**Comment by jzinner** [comment]
TODO next: 
Use the #acoustic channel - contact Tina to verify that these settings are configured correctly in the acoustic documentation: https://help.goacoustic.com/hc/en-us/articles/360043609093-Email-administration

**Comment by jzinner** [comment]
Per Stephmcv [Slack] - the donate system calls to a product called `Acoustic Transact`, (not `Accoustic Campaign`).  
The documentation for setting up BCC for Accoustic Transact is located at: https://developer.goacoustic.com/acoustic-campaign/docs/getting-started-with-transact-bcc 

Per Tina: [Slack]  there are not BCC settings in the Transact version of Acoustic.
This does not seem to be a simple [configuration] fix.

**Comment by jzinner** [comment]
Hi @danielfmiranda, another aspect Steph raised for this issue is validating we are sending well formatted XML to acoustic transact. 

Could you look in the donate-wagtail codebase and determine where / if an XML message is being created (or if we are in fact, passing json fields acoustic API & the acoustic system is transforming them into XML?)   

Thanks for your help exploring this.

**Comment by danielfmiranda** [comment]
Hi Everyone! 

I was able to find this function titled AcousticTransact in acoustic.py: https://github.com/mozilla/donate-wagtail/blob/77d42f00ff3d7e5a89c6bfdb277fa60285990d3d/donate/utility/acoustic/acoustic.py#L102 which seems to houses the **send_mail** function that is called whenever a user makes a donation here: https://github.com/mozilla/donate-wagtail/blob/77d42f00ff3d7e5a89c6bfdb277fa60285990d3d/donate/payments/tasks.py#L110

This function seems to do some work with XML before sending the email, is this what you were referencing? 

Let me know if you need anything else, thank you!

**Comment by stephaniemcv** [comment]
I wanted to update that I reached out to Acoustic Support and with the help of documentation they provided this bug has been fixed. Acoustic Transact is now back to sending emails to receipt-bcc@mozillafoundation.org in addition to the donor's email address.

I am creating [documentation](https://docs.google.com/document/d/1rTBakreXq23jrZ4soNYp4vqXSO2NPsybQj2UEr2EPWg/edit?usp=sharing) on the setup, as we are actually using the MyCopy feature rather than the Transact BCC. 

Thank you to everyone who helped investigate this weird bug.
